### PR TITLE
Update/vsts chapter

### PR DESCRIPTION
**Pull request by rtadd-git**
I have updated the old chapter on VSTS to now be a chapter about DevOps. In addition I have expanded content where I have something more to say and split out the section on work planning which I felt is a slightly standalone topic.

**Review request:** Please can you read the new content and let me know any parts where you think it is either unclear or incorrect. I am particularly interested to know whether you think there is anything else a new user would need to know to be able to get started with DevOps.

### CODE REVIEW

File: 03-devops-overview.Rmd [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+- [Bitbucket](https://bitbucket.org)
+- [GitLab](gitlab.com/)
+- [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/).
+
+We are using **Azure DevOps** as this is the Department's tool of choice for internal version control. We would use platforms such as GitHub for our more publicly facing projects (such as this book).
+
+To work with internal DevOps Repos you will need an account with Basic access level.
+There is a form to sort out DevOps account levels and access on the IT service portal - search for DevOps.
+You will also need someone to give you permission to see the DevOps project that you're working on.
+For information about how to do this, pleae read on.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:22]: type: pleae

File: 03-devops-overview.Rmd [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+- [GitHub](github.com)
+- [Bitbucket](https://bitbucket.org)
+- [GitLab](gitlab.com/)
+- [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/).
+
+We are using **Azure DevOps** as this is the Department's tool of choice for internal version control. We would use platforms such as GitHub for our more publicly facing projects (such as this book).
+
+To work with internal DevOps Repos you will need an account with Basic access level.
+There is a form to sort out DevOps account levels and access on the IT service portal - search for DevOps.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:20]: Can we link to IT service portal form? Or is that not a good idea in a public facing document?

File: 03-devops-overview.Rmd [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+The following diagram illustrates the concept of server and local copies. 
+
+![](www/distributed_version_control.png)
+
+## Creating a DevOps repository for a new project
+
+To set up a repository on DevOps you need to:
+
+1. Open the DevOps website.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:41]: A link to DevOps website?

File: 03-devops-overview.Rmd [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+2. Navigate to repositories.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:42]: Can we have a screenshot? With usernames blanked out if they are visible

File: 03-devops-overview.Rmd [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+It is easiest to do this by:
+
+1. Going to "Files" in the "Repos" section of DevOps.
+2. Ensuring you are in the right repository by checking the name in the banner at the top of the screen.
+3. Clicking the clone button in the top right-hand corner.
+4. Copying the URL.
+
+Once you have copied the URL, you need to open git bash on your computer and clone the repository using the URL.
+When you open git bash you should do so in the folder where you want the repo to be.
+Ideally this will be on your C drive, outside of OneDrive of Teams because git repositories don't work well if OneDrive
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:65]: typo: OneDrive of Teams

File: 03-devops-overview.Rmd [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+or Teams are syncing them.
+There is no need for the additional syncing from OneDrive or Teams because DevOps provides the online back-up that others can access.
+This means that any general rules than work should be on OneDrive or Teams to be backed up does not apply to repos because DevOps provides the back-up *as long as you push your work regularly and at least once a day*.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:68]: Maybe reword this? It's a bit confusing:

'As long as you push your work regularly and at least once a day, DevOps provides the back-up and other general rules that are usually applied to files on OneDrive or Teams are not necessary.'

File: 03-devops-overview.Rmd (hunk 4 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd:42]: I was thinking images would be a separate task so we can look at the whole book consistently in one go. Can you work with this?

File: 03-devops-overview.Rmd (hunk 3 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd:41]: Sadly not, I think that URL should be kept internal

File: 03-devops-overview.Rmd (hunk 2 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd:20]: As with the DevOps URL, I think not in a public document

File: 03-devops-overview.Rmd (hunk 6 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd]: I've gone for some hybrid of your suggestion and what I had before. I do really like your suggestion of bring the "as long as..." to the start rather than the end.

File: 03-devops-overview.Rmd (hunk 5 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd]: Should now be fixed

File: 03-devops-overview.Rmd (hunk 1 above)
Comment (rtadd-git) [reply 03-devops-overview.Rmd]: It should now be fixed

File: 03-devops-overview.Rmd [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+3. Clicking the clone button in the top right-hand corner.
+4. Copying the URL.
+
+Once you have copied the URL, you need to open git bash on your computer and clone the repository using the URL.
+When you open git bash you should do so in the folder where you want the repo to be.
+Ideally this will be on your C drive, outside of OneDrive or Teams, because git repositories don't work well if OneDrive
+or Teams are syncing them.
+There is no need for the additional syncing from OneDrive or Teams because DevOps provides the online back-up that others can access.
+*As long as you push your work regularly and at least once a day*, DevOps provides the back-up and any general rules than work should be on OneDrive or Teams to be backed up do not apply.
+
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:69]: This is ok for now, but we will have to check that we are not repeating ourselves from earlier chapters (I'm guessing cloning is covered to some degree in Chapter 2...?)

File: 03-devops-overview.Rmd [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+To add users to a project:
+
+1. In the Overview section, go to the summary page 
+2. Click on the invite button in the top, right hand corner
+2. Search for user
+3. Click Add
+
+The user will then get a welcome email inviting them to the project.
+
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:107]: Maybe add something about levels of access? Stakeholder vs. basic. This has caused some problems in my team because someone only had stakeholder access and was unable to clone a repo, took us a while to figure out what was going on.

File: 03-devops-overview.Rmd [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+## Pull Requests
+
+A pull request is a particularly useful bit of DevOps functionality.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:110]: I think it should be mentioned early on that pull requests are used when you are on a separate branch, prior to merging back into the main branch.

File: 03-devops-overview.Rmd [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+The visual interface allows you to:
+
+1. Give background to the changes made and link any work items
+2. Assign a colleague, or colleagues, to conduct a code review on the changes made
+3. Discuss the changes and resolve any requests for improvements
+4. Present the differences between the two branches in a visual manner (added/removed) so that you can easily see exactly what has been changed.
+
+As a pull request ensures someone reviews your code before it is incorporated into the main/master branch it is a good tool for quality assurance. Having another person look over your code before it is accepted reduces the risk of errors being missed.
+
+A good way of working involves all merging of new work is done by pull request to enfore the review process.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:123]: type - enfore
Also I think this could be worded better

File: 03-devops-overview.Rmd [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+1. Why you were doing the work you've done.
+2. Any context to the assumptions you've made whilst doing the work.
+3. A list of specific points that you would like your reviewer to check. This should always include that the code is clear enough and whether it requires further comments.
+
+Once you are happy with the code review that's taken place in a pull request:
+
+1. The person reviewing it approves the request to mark that the review is finished.
+2. The person who wrote the code and set the pull request up "completes" it which performs the merge.
+
+DevOps will delete the branch you were working in (the source brancg) once complete the pull request.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:137]: typo: branch

File: 03-devops-overview.Rmd [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+This is good practice because it keeps the repo tidy.
+Deleting branches after you have merged the work elsewhere does not mean you lose work because all that work is on the branch you've merged to.
+
+## Branch Policies
+
+At this point you will have a project that anyone who is a member of has the ability to push directly to the main/master branch. 
+
+Best practice ways of working include frequent code reviews v requests to support Quality Assurance.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:145]: Do you mean "code reviews and requests"?

File: 03-devops-overview.Rmd [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+Instead what we do is set **Branch Policies** which force code to be reviewed in DevOps before it can be merged into main/master.
+If you set any branch policies, changes to main/master have to be made using a pull request.
+
+### How to set branch policies
+
+To set branch policies:
+
+1. Go the the "Branched" page under repos in DevOps. Make sure you are looking at your intended repo by checking the centre top of the page.
+2. Hover your mouse over your main/master branch and click on the three vertical dots that appear at the right hand side of the screen.
+3. Clock on "Branch policies" in the menu that appears.
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:157]: Typo - click

File: 03-devops-overview.Rmd [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+
+### The branch policy options
+
+At the time of writing, branch policies offer four options:
+
+1. **Require a minimum number of reviewers** - how many people need to approve the code before it can be merged.
+2. **Check for linked work items** - Does the pull request need to have a work item linked
```
Comment (tmorley1) [inline 03-devops-overview.Rmd:164]: Not sure what you mean by a linked work item

**Review by tmorley1 — APPROVED** [review]
